### PR TITLE
refactor(forms): Clean up the find function.

### PR DESCRIPTION
**Pull request by dylhunn**
Move `find` into the `AbstractControl` hierarchy, and clean up a longstanding implementation todo.

### CODE REVIEW

File: packages/forms/src/model.ts [hunk 1]
```diff
@@ -999,7 +974,11 @@ export abstract class AbstractControl {
    * * `this.form.get(['items', 0, 'price']);`
    */
   get(path: Array<string|number>|string): AbstractControl|null {
```
Comment (alfaproject) [inline packages/forms/src/model.ts:976]: I guess this should be
```suggestion
  get(path: Array<string|number>|string|null|undefined): AbstractControl|null {
```

File: packages/forms/src/model.ts [hunk 2]
```diff
@@ -2511,4 +2500,9 @@ export class FormArray extends AbstractControl {
     control.setParent(this);
     control._registerOnCollectionChange(this._onCollectionChange);
   }
+
+  /** @internal */
+  override _find(name: string|number): AbstractControl|null {
+    return this.at(name as number) || null;
```
Comment (alfaproject) [inline packages/forms/src/model.ts:2506]:
```suggestion
    return this.at(name as number) ?? null;
```

File: packages/forms/src/model.ts (hunk 1 above)
Comment (dylhunn) [reply packages/forms/src/model.ts:976]: We do not want to change the existing method signature of `get` -- it should not be necessary to accept `null` or `undefined` as input.

File: packages/forms/src/model.ts (hunk 2 above)
Comment (dylhunn) [reply packages/forms/src/model.ts:2506]: Fixed.

**Review by jessicajaniuk — APPROVED** [review]
LGTM :cookie:

File: packages/forms/src/model.ts (hunk 1 above)
Comment (alfaproject) [reply packages/forms/src/model.ts:976]: Why check for null or undefined then? o:

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit 4c80ca3700e9a36922276d6cf420ac96103a486e.

**Pull request closed without merging**